examples/seq2seq/seq2seq_trainer.py

```diff
@@ -114,23 +123,31 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
                 else DistributedSampler(self.train_dataset)
             )
 
-    def compute_loss(self, model, inputs):
-        labels = inputs.pop("labels")
-        outputs = model(**inputs, use_cache=False)
-        logits = outputs[0]
-        return self._compute_loss(logits, labels)
-
-    def _compute_loss(self, logits, labels):
+    def _compute_loss(self, model, inputs):
         if self.args.label_smoothing == 0:
-            # Same behavior as modeling_bart.py
-            loss_fct = torch.nn.CrossEntropyLoss(ignore_index=self.config.pad_token_id)
-            assert logits.shape[-1] == self.vocab_size
-            loss = loss_fct(logits.view(-1, logits.shape[-1]), labels.view(-1))
+            # set all ids to -100 to be ignored
+            if self.data_args.ignore_pad_token_for_loss:
+                assert self.config.pad_token_id >= 0, "Make sure that `config.pad_token_id` is correcly defined"
+                inputs["labels"][inputs["labels"] == self.config.pad_token_id] = -100
+
+            # compute usual loss via models
+            loss, logits = model(**inputs)[:2]
```

Comment:
`use_cache` defaults to `None`, we won't need to set it to `False` explicitly

Reply:
If `config.use_cache=True` and the model is not BART, I suspect this will break.
I see no reason to let the config control a variable we know should be `False`

Reply:
This will never break because we never provide the actual `past` tensors.